rfcs/experimental/flow_graph_wait_try_put_and_wait/README.md

```diff
@@ -0,0 +1,604 @@
+in this section are for `multifunction_node` but also affects `async_node` in the same manner):
+
+```cpp
+
+using mf_node_type = multifunction_node<int, std::tuple<int, int>>;
+using output_ports = typename mf_node_type::output_ports_type;
+
+mf_node_type mfn(g, unlimited,
+    [](int input, output_ports& ports) {
+        std::get<0>(ports).try_put(output);
+        std::get<1>(ports).try_put(output);
+    })
+
+```
+
+Unlike the `function_node`, the computed output is not propagated implicitly to the successors. It is done explicitly by the user by calling
+`try_put(output)` on the corresponding output_port of the node. Since the metainformation associated with the particular input is hidden from the user
+inside of the Flow Graph implementation, it cannot be propagated to the successors as part of the explicit user try_put.
+
+An other interesting use-case is when the `multifunction_node` is used as a reduction for multiple input values with only one output. In that case the metainformation
+should not be automatically propagated at all and should be accumulated simultaneously with the outputs.
+
+Current implementation does not support ``multifunction_node`` and ``async_node``. 
+
+Possible approaches for implementing such support:
+
+* Hiding metainfo inside of ``output_ports`` to preserve automatic metainfo propagation.
+* Merging the input and the ``message_metainfo`` together into some publicly available type ``tagged_input<T>`` and require the user to explicitly specify this type
+for multi-output nodes. We can also consider propagating metainfo as part of ``tagged_input`` for each node type, not only the multi-output nodes.
+* Introducing an extra unspecified type ``node_type::hint_type`` and require the user to accept it as a third argument of the body. This option implies significant changes since
+  it makes ``message_metainfo`` public in some manner by expressing the ``hint_type`` including describing the safety guarantees for each action. Also, it would be needed to extend
+  ``output_ports::try_put`` function with the ``hint_type`` argument and potentially to extend ``receiver::try_put`` with such an argument.
+
+## Possible improvements for design and implementation
+
+### ``message_metainfo`` semantic changes
+
+Increment and decrement of the reference counter/s stored in the ``message_metainfo`` can be done in the RAII manner - increase while creating and copying the object and decreased
+while destroying the object.
+
+Extending this approach would allow implementing move semantics for ``message_metainfo`` and optimize passing it to the last successor of the buffering node or a ``continue_node`` without
+modifying the reference counter.
+
+## Exit criteria
+
+The following questions should be addressed before moving this feature to ``supported``:
+* Multi-output nodes support should be described and implemented.
```

Comment:
Add a link to the corresponding section of the document?

Reply:
Done